core/src/main/scala/io/gearpump/util/Graph.scala

```diff
@@ -254,6 +254,90 @@ class Graph[N, E](vertexList: List[N], edgeList: List[(N, E, N)]) extends Serial
   }
 
   /**
+   * Return all circles in graph.
+   */
+  def findCircles: mutable.Set[mutable.Set[N]] = {
+    val inStack = mutable.Map[N, Boolean]()
+    val stack = mutable.Stack[N]()
+    val DFN = mutable.Map[N, Int]()
```

Comment:
DFN, LOW is not good varible name. Maybe index and lowlink better? 
https://en.wikipedia.org/wiki/Tarjan%27s_strongly_connected_components_algorithm